packages/edit-site/src/components/sidebar-navigation-title/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalHeading as Heading,
+	__experimentalView as View,
+	__experimentalNavigatorBackButton as NavigatorBackButton,
+	__experimentalSpacer as Spacer,
+	Button,
+} from '@wordpress/components';
+import { isRTL, __ } from '@wordpress/i18n';
+import { chevronRight, chevronLeft } from '@wordpress/icons';
+
+export default function SidebarNavigationTitle( {
+	parentHref,
+	parentTitle,
+	title,
+} ) {
+	return (
+		<VStack spacing={ 10 }>
+			<View>
+				<HStack spacing={ 2 }>
+					{ ! parentHref && (
+						<NavigatorBackButton
+							style={
+								// TODO: This style override is also used in ToolsPanelHeader.
+								// It should be supported out-of-the-box by Button.
+								{
+									minWidth: 24,
+									padding: 0,
+									color: 'currentColor',
+								}
+							}
+							icon={ isRTL() ? chevronRight : chevronLeft }
+							isSmall
+							aria-label={ __( 'Navigate to the previous view' ) }
+						/>
+					) }
+					{ parentHref && (
+						<Button
+							style={
+								// TODO: This style override is also used in ToolsPanelHeader.
+								// It should be supported out-of-the-box by Button.
+								{
+									minWidth: 24,
+									padding: 0,
+									color: 'currentColor',
+								}
+							}
+							icon={ isRTL() ? chevronRight : chevronLeft }
+							href={ parentHref }
+							isSmall
+							aria-label={ __( 'Navigate to the previous view' ) }
+						></Button>
+					) }
+					<Spacer>
+						<Heading
+							level={ 5 }
+							style={ { color: 'currentColor' } }
```

Comment:
It's very hard to use our current components in a different context without tweaks like that. Would love if you folks can take a look and make our components more versatile and adapt automatically to their context.

For instance, why Heading, Text, Item have explicit colors? Shouldn't they just rely on cascade?

Reply:
Marco and I haven't been involved in the design of this component, but looking at the API I presume the point of the Text/Heading components is that it sets everything and doesn't rely on outer context. I imagine it would add another layer of complexity if some things cascaded as normal and some things didn't (size, line-height, etc).

Improving component versatility is almost all I've been working on, but everything is in bad shape so it's mostly a matter of priority. Good news is we are currently prioritizing color themeability (#44116), which should make it more straightforward in use cases like this.

Only suggestion here is to use the designated [API](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-heading--default):

```suggestion
							color="currentColor"
							style={ { color: 'currentColor' } }
```

Reply:
Ok. btw, it would be cool to take the existing admin schemes into consideration for the theming work to avoid having "two" competitive theming approaches in the same code base.

That said, I fail to see how theming would solve this because, in this case, the theme is I presume already defined by the user profiles of the WP user in general but I'd like to have a darker background for that same them in a the sidebar but keep a light background in the editor canvas...

Reply:
> why Heading, Text, Item have explicit colors? Shouldn't they just rely on cascade?
> It's very hard to use our current components in a different context without tweaks like that. Would love if you folks can take a look and make our components more versatile and adapt automatically to their context

With regards to the APIs and behavior or text components (`Heading`, `Text`, `Item`...), Lena already gave a great explanation to why they may behave like that. Their whole point seems to be to ensure a predictable text style, which also includes text color.

More in general, I also agree with Lena that the vast majority of our time is spent _exactly_ on improving the component library, on reducing its technical debt and more in general on making it easier to use for its consumers (developers). The rest of the time is spent fixing bugs and regressions, mostly caused by changes usually caused by external contributors (usually caused by rushed decisions and poor testing).

The thing is, there's just _a lot_ of work that got accumulated through the years, mostly caused by lack of a coherent API design and tight coupling between the components and the gutenberg app code.

> Ok. btw, it would be cool to take the existing admin schemes into consideration for the theming work to avoid having "two" competitive theming approaches in the same code base.

The idea is that the components library exposes its own theming variables, in order to have full control on how these variables are used / exposed, and in order to avoid to have the components library being tightly coupled to WordPress' admin schemes (although we're still using the admin scheme colors as a fallback). This way, the components library will be easier to use and theme when used outside of the gutenberg project.

You can find more details of the approach that we're experimenting with in this dedicated issue: https://github.com/WordPress/gutenberg/issues/44116 — although please bear in mind that we're still in a very early phase and we're still experimenting with its core APIs.

> That said, I fail to see how theming would solve this because, in this case, the theme is I presume already defined by the user profiles of the WP user in general but I'd like to have a darker background for that same them in a the sidebar but keep a light background in the editor canvas...

One way theming could help here is by exposing (in the future) a variable for the foreground (text) color.

Because the Theme variables can be nested / overridden, it will be possible to override the foreground color for specific React subtrees.

Reply:
Yes, the requirements we have in mind are:

- Works with or without the WP admin scheme system (though only the primary  admin color is relevant in Gutenberg)
- Allows themes to be scoped to a specific section of the page
- Allows themes to be nested (not all theme color values need to be specified as they will cascade)
- Has to work with both dark and light background color values
- Can be overridden without React if necessary

The basic API we're working toward at the moment is like:

```jsx
<WhateverDarkSidebar>
  <Theme background="#000">
    <WhateverComponents />
  </Theme>
</WhateverDarkSidebar>
```

And `<Theme>` will render as a div with the color values (as CSS variables) [scoped](https://github.com/WordPress/gutenberg/issues/44116#issuecomment-1245550772) to it. We're hoping we can generate the text/border colors automatically based on the given background color.

Reply:
Ok, thanks for the details both of you. I think that's a good list of requirements. I'd personally add 

 - components outside the components package should be able to be themable, so consume the theme's config some how. 
 - I'd try to limit the config options to the minimum personally as well to avoid exponential complexity.